File: Arrays and Hasing/groupAnagram.py
from collections import defaultdict
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        
        holder = defaultdict(list)
        
        for str in strs:
            counter = [0] * 26
            for c in str:
                counter[ord(c) - ord('a')]+=1
            
            holder[tuple(counter)].append(str)
            
        result = []    
        for key, value in holder.items():
            result.append(value)
            
        return result

File: Arrays and Hasing/groupAnagram.cpp
#include <vector>
#include <string>
#include <unordered_map>

class Solution {
    public:
        std::vector<std::vector<std::string>> groupAnagrams(std::vector<std::string>& strs) {
            
            std::unordered_map<std::string, std::vector<std::string>> holder;
            
            //we need to look ar every single string in the vector list

            for(const auto& str : strs){
                std::vector<int> count(26,0);

                for(auto c : str){
                    count[c - 'a']++;
                }

                std::string key = std::to_string(count[0]);
                for(int i = 0; i < 26; i++){
                    key+= ',' + std::to_string(count[i]);
                }
                holder[key].push_back(str);
            }

            std::vector<std::vector<std::string>> result;
            for(const auto& pair : holder){
                result.push_back(pair.second);
            }
            return result;
        }
    };
    

    int main(){

        Solution solu;
        

        return 0;
    }